pydm/widgets/analog_indicator.py

```diff
@@ -0,0 +1,1576 @@
+        """
+        self.set_position()
+        self.repaint()
+
+    def set_value(self, value):
+        """
+        Set a new current value for the indicator.
+        """
+        self._value = value
+        self.update_indicator()
+
+    def set_upper_limit(self, new_limit):
+        """
+        Set the scale upper limit.
+
+        Parameters
+        ----------
+        new_limit : float
+            The upper limit of the scale.
+        """
+        self._upper_limit = new_limit
+
+    def set_lower_limit(self, new_limit):
+        """
+        Set the scale lower limit.
+
+        Parameters
+        ----------
+        new_limit : float
+            The lower limit of the scale.
+        """
+        self._lower_limit = new_limit
+
+    def set_upper_minor_alarm(self, new_minor_alarm):
+        """
+        Set the scale upper minor alarm.
+
+        Parameters
+        ----------
+        new_minor_alarm : float
+            The upper minor alarm of the scale.
+        """
+        self._upper_minor_alarm = new_minor_alarm
+
+    def set_lower_minor_alarm(self, new_minor_alarm):
+        """
+        Set the scale lower minor alarm.
+
+        Parameters
+        ----------
+        new_minor_alarm : float
+            The lower minor alarm of the scale.
+        """
+        self._lower_minor_alarm = new_minor_alarm
+
+    def set_upper_major_alarm(self, new_major_alarm):
+        """
+        Set the scale upper major alarm.
+
+        Parameters
+        ----------
+        new_major_alarm : float
+            The upper major alarm of the scale.
+        """
+        self._upper_major_alarm = new_major_alarm
+
+    def set_lower_major_alarm(self, new_major_alarm):
+        """
+        Set the scale lower major alarm.
+
+        Parameters
+        ----------
+        new_major_alarm : float
+            The lower major alarm of the scale.
+        """
+        self._lower_major_alarm = new_major_alarm
+
+    def get_show_ticks(self):
+        return self._show_ticks
+
+    def set_show_ticks(self, checked):
+        if self._show_ticks != bool(checked):
+            self._show_ticks = checked
+            self.repaint()
+
+    def get_orientation(self):
+        return self._orientation
+
+    def set_orientation(self, orientation):
+        self._orientation = orientation
+        self.adjust_transformation()
+        self.repaint()
+
+    def get_flip_scale(self):
+        return self._flip_scale
+
+    def set_flip_scale(self, checked):
+        self._flip_scale = bool(checked)
+        self.adjust_transformation()
+        self.repaint()
+
+    def get_inverted_appearance(self):
+        return self._inverted_appearance
+
+    def set_inverted_appearance(self, inverted):
+        self._inverted_appearance = inverted
+        self.adjust_transformation()
+        self.repaint()
+
+    def get_bar_indicator(self):
+        return self._barIndicator
+
+    def set_bar_indicator(self, checked):
```

Comment:
I couldn't figure out what the bar indicator setting did- everything else worked as expected!